src/mojo-serial/MojoSerial/PluginSiPixelRecHits/GPUPixelRecHits.mojo

```diff
@@ -0,0 +1,227 @@
+from math import sqrt
+from utils.numerics import max_finite
+
+from MojoSerial.DataFormats.BeamSpotPOD import BeamSpotPOD
+from MojoSerial.DataFormats.ApproxAtan2 import ApproxAtan2
+from MojoSerial.CUDADataFormats.TrackingRecHit2DSOAView import (
+    TrackingRecHit2DSOAView,
+)
+from MojoSerial.Geometry.Phase1PixelTopology import AverageGeometry
+
+import MojoSerial.CUDADataFormats.TrackingRecHit2DHeterogeneous as TrackingRecHit2DHeterogeneous
+import MojoSerial.CUDADataFormats.SiPixelDigisSoA as SiPixelDigisSoA
+import MojoSerial.CUDADataFormats.SiPixelClustersSoA as SiPixelClustersSoA
+import MojoSerial.CondFormats.PixelCPEforGPU as PixelCPEforGPU
+
+
+fn getHits(
+    cpeParams: UnsafePointer[PixelCPEforGPU.ParamsOnGPU],
+    bs: UnsafePointer[BeamSpotPOD],
+    pdigis: UnsafePointer[SiPixelDigisSoA.DeviceConstView],
+    numElements: UInt32,
+    pclusters: UnsafePointer[SiPixelClustersSoA.DeviceConstView],
+    phits: UnsafePointer[TrackingRecHit2DSOAView],
+):
+    # FIXME
+    # the compiler seems NOT to optimize loads from views (even in a simple test case)
+    # The whole gimnastic here of copying or not is a pure heuristic exercise that seems to produce the fastest code with the above signature
+    # not using views (passing a gazzilion of array pointers) seems to produce the fastest code (but it is harder to mantain)
+
+    debug_assert(phits)
+    debug_assert(cpeParams)
+
+    ref hits = phits[]
+
+    ref digis = pdigis[]  # the copy is intentional!
```

Comment:
This does not seem to match the comments above and behind the declaration; `digis` is a reference not a copy?